wave_vision/include/wave/vision/dataset.hpp

```diff
@@ -32,37 +35,42 @@ namespace wave {
  */
 class VOTestCamera {
  public:
-    int image_width;
-    int image_height;
-    Mat3 K;
-    double hz;
+    int image_width = 0;
+    int image_height = 0;
+    Mat3 K = Mat3::Identity();
+    double hz = 0.0;
+
+    double dt = 0.0;
+    int frame = 0;
 
-    double dt;
-    int frame;
+    VOTestCamera() {}
 
-    VOTestCamera()
-        : image_width{-1}, image_height{-1}, K{}, hz{-1}, dt{0}, frame{-1} {}
+    VOTestCamera(int image_width, int image_height, Mat3 K, double hz)
+        : image_width{image_width}, image_height{image_height}, K{K}, hz{hz} {}
 
     /** Check whether camera is triggered at this time-step
      * @returns boolean to denote true or false
      */
     bool update(double dt);
 
-    /** Returns a list of observable 3D world features
+    /** Gives a list of measurements of visible landmarks, if enough time has
+     * passed since the last frame
      *
      * @param dt Update time step
-     * @param features 3D world features to test (NWU)
-     * @param rpy Vector containing roll, pitch, yaw of camera (EDN)
-     * @param t Vector containining position in x, y and z (EDN)
+     * @param landmarks map of landmarks with 3D position in world frame
+     * @param q_GC orientation of camera in global frame
+     * @param G_p_GC translation to camera from origin, in global frame
```

Comment:
I intentionally added EDN, NWU to be very specific. In ROS for example the landmarks could be defined in ENU rather than NWU for example.